relay-general/src/store/normalize.rs

```diff
@@ -293,15 +320,16 @@ fn remove_invalid_measurements(
         }
 
         // Retain payloads in _meta just for excessive custom measurements.
-        if let Some(measurement) = value.value_mut().take() {
-            removed_measurements.insert(name.clone(), Annotated::new(measurement));
-        }
+        removed_measurements.insert(name.clone(), Annotated::new(std::mem::take(measurement)));
 
         false
     });
 
-    if !removed_measurements.is_empty() {
+    if custom_measurements_count >= measurements_config.max_custom_measurements {
```

Comment:
Being at the max should not be a problem?
```suggestion
    if custom_measurements_count > measurements_config.max_custom_measurements {
```

Reply:
if custom_measurements_count == max_custom_measurements, it will not pass the `if-block` at line 317 so it will be removed, thats why I made it like this, however I realize now, if it reaches the same num as max_cusotm_measurements on the final iteration, it will think it removed a measurement even when it didn't.

so I'll change this to rather be a flag